pkg/plugins/gateway/algorithms/router.go

```diff
@@ -31,6 +34,8 @@ var routerConstructor = map[types.RoutingAlgorithm]types.RouterProviderRegistrat
 
 // Validate validates if user provided routing routers is supported by gateway
 func Validate(algorithms string) (types.RoutingAlgorithm, bool) {
+	routerMu.RLock()
```

Comment:
due to data race condition for router factory variable?

Reply:
Yes, this ensures thread-safe concurrent R/W and prevents data race conditions.